public/js/actions/breakpoints.js

```diff
@@ -114,6 +115,25 @@ function removeAllBreakpoints() {
   };
 }
 
+function toggleAllBreakpoints(shouldDisableBreakpoints) {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    const breakpoints = getBreakpoints(getState());
+    return dispatch({
+      type: constants.TOGGLE_BREAKPOINTS,
+      shouldDisableBreakpoints,
+      [PROMISE]: (async function () {
+        for (let [, breakpoint] of breakpoints) {
```

Comment:
Yeah, I think the destructuring is fine. However, you could send off the requests in parallel:

`[PROMISE]: Promise.all(breakpoints.map(bp => dispatch(...)))`